scaladoc/src/dotty/tools/scaladoc/site/StaticSiteContext.scala

```diff
@@ -71,6 +71,19 @@ class StaticSiteContext(
     orphanedFiles.flatMap(p => loadTemplate(p.toFile, isBlog = false))
   }
 
+  lazy val redirectTemplates: Seq[LoadedTemplate] = {
+    def doFlatten(t: LoadedTemplate): Seq[LoadedTemplate] =
+      t +: t.children.flatMap(doFlatten)
+    val mainFiles = templates.flatMap(doFlatten)
+    mainFiles.flatMap { loadedTemplate =>
+      loadedTemplate.templateFile.settings.apply("page").asInstanceOf[Map[String, Object]].get("redirectFrom").map { case redirectFrom: String =>
```

Comment:
Lack of type safety is coming from `liquid templates` which use java `Map<String, Object>` for storing values properties. This operator looks nice, though I don't know if it's worth introducing it now. To be honest, I feel the scaladoc needs rewrite (it was prototyped in less than 2 months and many things were pushed too fast), or at least some components like `static site`, from scratch.

Reply:
Changing it to a list should be an easy fix

Reply:
> To be honest, I feel the scaladoc needs rewrite (it was prototyped in less than 2 months and many things were pushed too fast), or at least some components like `static site`, from scratch.

This is interesting to know. Would you mind creating a list of issues to keep track of the tasks that should be performed?

Reply:
I mean, this is just my hot take on some problems working with a static site, though I still feel not so proud of some things that are still in the codebase (there are tons of TODOs,  for example, we have poor error handling because many things have been introduced before we could attach to context and use a proper reporter, so they are just printlns or uncaught exceptions). Nonetheless, to revisit the scaladoc and think what should be rewritten we need some more time, which I lack at the moment, so it won't happen until next year I think.

Reply:
I think we can’t afford much technical debt on scaladoc if we want it to reliably produce the documentation of the Scala ecosystem. So, I really encourage you to take some time to file issues describing what could be improved in the codebase to make scaladoc easier to maintain and more robust.